console-subscriber/src/builder.rs

```diff
@@ -14,6 +14,54 @@ use tracing_subscriber::{
     registry::LookupSpan,
 };
 
+/// A subscriber that discards everything
+///
+/// NoSubscriber seems like it would do that, but setting it for a
+/// single thread won't have the desired effect.  The dispatch logic
+/// uses NoSubscriber (through Dispatcher::none()) as a default and as a
+/// sentinel value.  When dispatching, it checks the thread-local
+/// subscriber using Any::is<NoSubscriber>, and has the global
+/// subscriber take over the thread-local subscriber if so.
+#[derive(Default)]
+struct NoNoSubscriber(tracing_core::subscriber::NoSubscriber);
+
+impl tracing_core::subscriber::Subscriber for NoNoSubscriber {
+    fn register_callsite(
+        &self,
+        metadata: &'static tracing::Metadata<'static>,
+    ) -> tracing_core::Interest {
+        self.0.register_callsite(metadata)
+    }
+
+    fn new_span(&self, span: &tracing_core::span::Attributes<'_>) -> tracing_core::span::Id {
+        self.0.new_span(span)
+    }
+
+    fn event(&self, event: &tracing::Event<'_>) {
+        self.0.event(event)
+    }
+
+    fn record(&self, span: &tracing_core::span::Id, values: &tracing_core::span::Record<'_>) {
+        self.0.record(span, values)
+    }
+
+    fn record_follows_from(&self, span: &tracing_core::span::Id, follows: &tracing_core::span::Id) {
+        self.0.record_follows_from(span, follows)
+    }
+
+    fn enabled(&self, metadata: &tracing::Metadata<'_>) -> bool {
+        self.0.enabled(metadata)
+    }
+
+    fn enter(&self, span: &tracing_core::span::Id) {
+        self.0.enter(span)
+    }
+
+    fn exit(&self, span: &tracing_core::span::Id) {
+        self.0.exit(span)
+    }
+}
```

Comment:
this issue was fixed upstream in [`tracing-core` 0.1.24`](https://github.com/tokio-rs/tracing/releases/tag/tracing-core-0.1.24), so if we change the `console-subscriber` crate's minumum `tracing-core` dependency to 0.1.24, we can get rid of this workaround.